drivers/adc/ad7606/ad7606.c

```diff
@@ -685,6 +687,55 @@ int32_t ad7606_read_samples(struct ad7606_dev *dev, uint32_t * data,
 	return 0;
 }
 
+/***************************************************************************//**
+ * @brief Convert the samples into human readable values using the ADC transfer
+ * function defined in the datasheet of the AD7606.
+ *
+ * @param dev        - The device structure.
+ * @param sample     - Raw sample to convert
+ * @param ch         - Pointer to location of buffer where to store the data.
+ *
+ * @return ret - voltage sample scaled in millivolts according to channel config
+*******************************************************************************/
+int32_t ad7606_scale_sample(struct ad7606_dev *dev, uint32_t sample,
+			    uint32_t ch)
+{
+	/* FIXME: this transfer function works fine for most AD7606 chips;
+	 *        some quirks may need fixing
+	*/
+	struct ad7606_range *range = &dev->range_ch[ch];
+	int64_t volt_range;
+	int64_t refinout = dev->refinout;
+	const int64_t voltage_div = 1000;
+	const uint32_t bits = ad7606_chip_info_tbl[dev->device_id].bits;
+	int64_t voltage = no_os_sign_extend32(sample, (bits - 1));
+
+	switch (dev->device_id) {
+	case ID_AD7606C_18:
+		volt_range = range->max;
+		voltage = (volt_range *
+			   voltage)
+			  /
+			  (262144 >> 1);
+		break;
+	default:
+		volt_range = range->max;
+		voltage = (voltage *
+			   volt_range * /* volt range value is either 2.5, 5 or 10V */
```

Comment:
are all these comments necessary? instead I'd use `_mv` prefix on the variables. I think it is suggestive enough.